info/available-rules.md

```diff
@@ -1,131 +1,131 @@
+| 1    | 1.2.1    | PACKAGE_NAME_INCORRECT_PATH               | Check: warns if the path for a file does not match with a package name<br>Fix: replacing the incorrect package name with the name constructed from a path to the file.                                                                                           | yes | no                                                                                                                                                                                                                         | Make this check isolated from domain name creation.Recursively update all imports in the project.Fix the directory where the code is stored.Add a test mechanism to test checker.                                                 |
+| 1    | 1.2.1    | INCORRECT_PACKAGE_SEPARATOR               | Check: warns if the underscore is incorrectly used to split the package naming<br>Fix: fixing all nodes in AST and the package name to remove all underscores                                                                                                        | no  | no                                                                                                                                                                                                                         | Recursively update usages of this class in the project                                                                                                                                                                        |
+| 1    | 1.3.1    | CLASS_NAME_INCORRECT                      | Check: warns if the Class/Enum/Interface name does not match Pascal case ("([A-Z][a-z0-9]+)+")<br>Fix: fixing the case: if it is some fixed case (like snake/camel) - with word saving; if not - will restore PascalCase as is.                          | yes | no                                                                                                                                                                                                                         | Recursively update usages of this class in the projectCheck and decide the better way of converting the identifier to PascalCaseNeed to add checks using natural language processing to check that the class name contains only nouns |
```

Comment:
Unclear sentence: "If it is some fixed case (like snake/camel) - with word saving; if not - will restore PascalCase as is." Please, explain.

Reply:
[`PascalCase`, `camelCase` and `snake_case` explained](https://khalilstemmler.com/blogs/camel-case-snake-case-pascal-case/).

Reply:
> Fix: fixing the case: if it is some fixed case (like snake/camel) - with word saving

This means, if the case is recognized to be a known one (Pascal, Camel or Snake), then we can detect word boundaries within the identifier name, and the fix is 100 percent accurate:

* `UPPER` -> `Upper`
* `lower` -> `Lower`
* `lowerCamel` -> `LowerCamel`
* `lower_snake` -> `LowerSnake`

> if not - will restore PascalCase as is

If the name of an identifier conforms to neither of the above 4 patterns, then a heuristic algorithm is applied:

* the 1st character, if lower-case, is capitalized: `c` -> `C`
* the character after an underscore is capitalized, and the underscore removed: `_c` -> `C`
* any upper-case character which follows another upper-case one is converted to the lower case: `AB` -> `Ab`

To summarize the above, an identifier named `paSCa_l` will be renbamed to `PaScaL`.

Reply:
OK. I slightly changed it:

Reply:
If it is some fixed case (like Snake or Camel) - with word saving; if not - will restore PascalCase as is.